x/upgrade/module.go

```diff
@@ -104,7 +104,9 @@ func (am AppModule) RegisterServices(cfg module.Configurator) {
 }
 
 // InitGenesis is ignored, no sense in serializing future upgrades
-func (am AppModule) InitGenesis(_ sdk.Context, _ codec.JSONCodec, _ json.RawMessage) []abci.ValidatorUpdate {
+func (am AppModule) InitGenesis(ctx sdk.Context, _ codec.JSONCodec, _ json.RawMessage) []abci.ValidatorUpdate {
+	version := am.keeper.GetProtocolVersion(ctx)
```

Comment:
Currently x/upgrade doesn't have a genesis state. I think this line will always return 0. We should probably add a `genesis.proto` and return a DefaultGenesis for x/upgrade

Reply:
hmm this may be wrong then, im not sure where to set this data for snapshot restoring. After we restore the snapshot we have to update the variable in BaseApp.

Reply:
this doesn't work actually. I think the only way this works if its part of BaseApp, not upgrade module. It would be a circular dependency though